### PR TITLE
Add burnWithSigner function

### PR DESCRIPTION
**Pull request by RenatoChitolina**
# Summary
### Added
- Added `burnWithSigner` function to enable burn NFTs with l2signer

### Changed
- mark `burn` as deprecated

# Why the changes
This is one of the series changes that we do to replace the **starkwallet** to **l2signer** to sign the message. 
the previous reference PR is: https://github.com/immutable/imx-core-sdk/pull/78

# Things worth calling out
Tested the function from this **imx-testing** repo: https://github.com/immutable/imx-testing/pull/78

### CODE REVIEW

File: src/workflows/burn.ts [hunk 1]
```diff
@@ -7,10 +7,16 @@ import {
...
 
+type burnWorkflowWithSignerRequest = WalletConnection & {
+  request: GetSignableBurnRequest;
+  transfersApi: TransfersApi;
+};
+
+/** @deprecated */
 export async function burnWorkflow(
-  signer: Signer,
+  signer: L1Signer,
```
Comment (carmen0208) [inline src/workflows/burn.ts:19]: let's keep the type of the deprecated function

**Review by carmen0208 — COMMENTED** [review]
LGTM. Very trivial comments. please also make sure you also make the changes on the imx-testing repo and pass all the test.

File: src/workflows/burn.ts (hunk 1 above)
Comment (yundi-fu) [reply src/workflows/burn.ts:19]: yes

Comment (RenatoChitolina) [reply src/workflows/burn.ts]: My bad, I did that for a purpose I changed after and forgot to put it back in the original state.
Done.

**Review by yundi-fu — APPROVED** [review]

File: src/workflows/burn.ts [hunk 2]
```diff
@@ -67,6 +73,62 @@ export async function burnWorkflow(
   };
 }
 
+export async function burnWorkflowWithSigner({
```
[the remark this answers is not in the document]
Comment (RenatoChitolina) [reply src/workflows/burn.ts:76]: That sounds like a good idea! @zatlite could you give permission to visualise the file you shared? I'm afraid I don't have access

[the remark this answers is not in the document]
Comment (zatlite) [reply src/workflows/burn.ts:76]: Sorry about that. Will see about the access later. 
The go code looks like this.

```
// BurnAddress 0x00 where tokens are sent to be removed from circulation
const BurnAddress = "0x0000000000000000000000000000000000000000"

// Burn permanently removes tokens from circulation.
// To burn an asset, you need to transfer the asset to the burn address.
// We have added burn functions in the SDK for clarity, and these are just wrappers for standard transfer functions to BurnAddress.
// Using the transfer functions with the burn address as a recipient will also work just the same.
func Burn(
	ctx context.Context,
	api *client.ImmutableXAPI,
	l1signer signers.L1Signer,
	l2signer signers.L2Signer,
	request types.GetSignableBurnRequest,
) (*models.CreateTransferResponseV1, error) {
	burnAddress := BurnAddress
	transferRequest := models.GetSignableTransferRequestV1{
		Amount:   &request.Amount,
		Sender:   &request.Sender,
		Token:    request.Token,
		Receiver: &burnAddress,
	}
	return transfersWorkflow.CreateTransfer(ctx, api, l1signer, l2signer, transferRequest)
}

**Review by zatlite — APPROVED** [review]

**Pull request merged**